### PR TITLE
add nil check of showonly interface value

### LINKED ISSUE #949: Crash when setting show_only to single empty string on helm_template
### Terraform, Provider, Kubernetes and Helm Versions

```
Terraform version:
Provider version:
Kubernetes version:
```

### Affected Resource(s)

- helm_template

### Terraform Configuration Files
```hcl
data "helm_template" "test" {
  show_only    = [""]
...
}
```

### Debug Output

NOTE: In addition to Terraform debugging, please set HELM_DEBUG=1 to enable debugging info from helm.

### Panic Output

https://gist.github.com/alexsomesan/033d7bff85011ab823c981176682843a

### Steps to Reproduce

1. `terraform apply`

### References

- GH-1234

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by BBBmau**
### Description

fix a crash when setting `show_only`  value to an empty string

Fixes #949 

### Acceptance tests
- [x] Have you added an acceptance test for the functionality being added?

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-helm/blob/main/CHANGELOG.md):

```release-note
* Fix show_only crash in helm_template
```
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: helm/data_template.go [hunk 1]
```diff
@@ -375,7 +375,9 @@ func dataTemplateRead(ctx context.Context, d *schema.ResourceData, meta interfac
 		showOnlyAttrValue := showOnlyAttr.([]interface{})
 
 		for _, showFile := range showOnlyAttrValue {
-			showFiles = append(showFiles, showFile.(string))
+			if showFile != nil {
+				showFiles = append(showFiles, showFile.(string))
+			}
```
Comment (jrhouston) [inline helm/data_template.go]: Something you could also do here is use the type assertion to test this, as if it's nil then trying to assert it as a string will fail:

```go

if s, ok := showFile.(string); ok {
  showFiles = append(showFiles, s)
}
```

Comment (BBBmau) [reply helm/data_template.go]: This definitely is a much better way to do it. Thanks for the recommendation!

File: helm/data_template.go [hunk 2]
```diff
@@ -375,7 +375,9 @@ func dataTemplateRead(ctx context.Context, d *schema.ResourceData, meta interfac
 		showOnlyAttrValue := showOnlyAttr.([]interface{})
 
 		for _, showFile := range showOnlyAttrValue {
-			showFiles = append(showFiles, showFile.(string))
+			if s, ok := showFile.(string); ok {
```
Comment (alexsomesan) [inline helm/data_template.go]: How about also adding a length check to exclude empty strings?
```suggestion
			if s, ok := showFile.(string); ok && len(s) > 0 {
```

Comment (BBBmau) [reply helm/data_template.go]: I see what you mean, we would not only check whether there exists the showFile but we would also exclude any strings that are empty

**Review by alexsomesan — APPROVED** [review]
LGTM. Hopefully this does the trick
